src/Core/Impl/Extensions/EnumerableExtensions.cs

```diff
@@ -89,18 +89,17 @@ public static IEnumerable<T> Ordered<T>(this IEnumerable<T> source)
             }
         }
 
-        public static IEnumerable<T> TraverseBreadthFirst<T>(this T root, Func<T, IEnumerable<T>> selectChildren) {
-            var items = new Queue<T>();
-            items.Enqueue(root);
-            while (items.Count > 0) {
+        public static IEnumerable<T> TraverseBreadthFirst<T>(this T root, Func<T, IEnumerable<T>> selectChildren)
+            => Enumerable.Repeat(root, 1).TraverseBreadthFirst(selectChildren);
+
+        public static IEnumerable<T> TraverseBreadthFirst<T>(this IEnumerable<T> roots, Func<T, IEnumerable<T>> selectChildren) {
+            var items = new Queue<T>(roots);
+
+            while (!items.IsNullOrEmpty()) {
                 var item = items.Dequeue();
                 yield return item;
 
-                var children = selectChildren(item);
-                if (children == null) {
-                    continue;
-                }
-
+                var children = selectChildren(item).MaybeEnumerate();
```

Comment:
Please don't use `MaybeEnumerate`. I'm going to remove it in future, so let's leave it as it was.